crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -150,6 +150,24 @@ def compute_resources_available?
       true
     end
 
+    # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+    def ha_deployed?
+      unless addon_installed? "ha"
+        errors.add(:base, I18n.t("api.crowbar.ha_not_installed"))
```

Comment:
do we plan to use this error somewhere? I think we should only write to the errors hash if we can't further process the instance of this object, not on every check.